apps/prairielearn/src/pages/instructorRequestCourse/instructorRequestCourse.html.ts

```diff
@@ -0,0 +1,256 @@
+import { HtmlSafeString, html } from '@prairielearn/html';
+import { renderEjs } from '@prairielearn/html-ejs';
+import { z } from 'zod';
+import { CourseRequestSchema, UserSchema } from '../../lib/db-types';
+
+export const CourseRequestRowSchema = z.object({
+  course_request: CourseRequestSchema,
+  approved_by_user: UserSchema.nullable(),
+});
+type CourseRequestRow = z.infer<typeof CourseRequestRowSchema>;
+
+export function RequestCourse({
+  rows,
+  resLocals,
+}: {
+  rows: CourseRequestRow[];
+  resLocals: Record<string, any>;
+}) {
+  return html`
+    <!doctype html>
+    <html lang="en">
+      <head>
+        ${renderEjs(__filename, "<%- include('../partials/head')%>", {
+          ...resLocals,
+        })}
+        <script>
+          $(function () {
+            $('input[name=cr-role]').change(function () {
+              var role = this.value;
+              $('.question-form button').prop('disabled', role != 'instructor');
+              $('.role-comment').hide();
+              $('.role-comment-' + role).show();
+            });
+          });
+        </script>
+      </head>
+      <body>
+        ${renderEjs(__filename, "<%- include('../partials/navbar')%>", {
+          ...resLocals,
+          navPage: 'request_course',
+        })}
+        <main id="content" class="container">
+          ${CourseRequestsCard({ rows })}
+          ${CourseNewRequestCard({ csrfToken: resLocals.__csrf_token })}
+        </main>
+      </body>
+    </html>
+  `.toString();
+}
+
+function CourseRequestsCard({ rows }: { rows: CourseRequestRow[] }): HtmlSafeString {
+  if (rows.length === 0) {
+    return html``;
```

Comment:
Nit: you can return a plain empty string (`''`) and change the return type to `HtmlValue`. This avoids an extra object allocation. I acknowledge that this basically doesn't matter here, but it's a helpful pattern to establish for higher-traffic pages.

Reply:
Done in af932a8.